index.js

```diff
@@ -1,105 +1,128 @@
-const options = {};
-
-['watch', 'clean'].forEach(key => {
-  const index = args.indexOf(`--${key}`);
-  if (index >= 0) {
-    options[key] = true;
-    args.splice(index, 1);
-  }
-});
-
-if (args.length < 2) {
-  console.error('Not enough arguments: copy-and-watch [options] <sources> <target>'.red);
-  process.exit(1);
+
+let watch = false, clean = false;
+const sourceGlobs = [];
+
+for (let arg of args) {
+	if (arg === '--watch') {
+		watch = true;
+	} else if (arg === '--clean') {
+		clean = true;
+	} else {
+		sourceGlobs.push(path.normalize(arg));
+	}
 }
 
-const target = args.pop();
-const sources = args;
-const parents = [...new Set(sources.map(globParent))];
-
-const findTarget = from => {
-  const parent = parents
-    .filter(p => from.indexOf(p) >= 0)
-    .sort()
-    .reverse()[0];
-  return path.join(target, path.relative(parent, from));
-};
-const createDirIfNotExist = to => {
-  'use strict';
-
-  const dirs = [];
-  let dir = path.dirname(to);
-
-  while (dir !== path.dirname(dir)) {
-    dirs.unshift(dir);
-    dir = path.dirname(dir);
-  }
-
-  dirs.forEach(dir => {
-    if (!fs.existsSync(dir)) {
-      fs.mkdirSync(dir);
-    }
-  });
-};
-const copy = from => {
-  const to = findTarget(from);
-  createDirIfNotExist(to);
-  const stats = fs.statSync(from);
-  if (stats.isDirectory()) {
-    return;
-  }
-  fs.writeFileSync(to, fs.readFileSync(from));
-  console.log('[COPY]'.yellow, from, 'to'.yellow, to);
-};
-const remove = from => {
-  const to = findTarget(from);
-  fs.unlinkSync(to);
-  console.log('[DELETE]'.yellow, to);
-};
-const rimraf = dir => {
-  if (fs.existsSync(dir)) {
-    fs.readdirSync(dir).forEach(entry => {
-      const entryPath = path.join(dir, entry);
-      if (fs.lstatSync(entryPath).isDirectory()) {
-        rimraf(entryPath);
-      } else {
-        fs.unlinkSync(entryPath);
-      }
-    });
-    fs.rmdirSync(dir);
-  }
-};
-
-// clean
-if (options.clean) {
-  rimraf(target);
+destination = sourceGlobs.pop(); //pick last path as destination
+
+if (sourceGlobs.length === 0 || destination === undefined) {
+	console.error('Not enough arguments: copy-and-watch [options] <sources> <target>'.red);
+	process.exit(1);
+}
+
+parents = [...new Set(sourceGlobs.map(globParent).map(path.normalize))];
+
+if (clean) {
+	console.log('Cleaning...');
+	rimraf(destination);
 }
 
 // initial copy
-sources.forEach(s => glob.sync(s).forEach(copy));
+sourceGlobs.forEach(s => glob.sync(s).forEach(copy));
 
 // watch
-if (options.watch) {
-  chokidar.watch(sources, {
-    ignoreInitial: true
-  })
-    .on('ready', () => sources.forEach(s => console.log('[WATCH]'.yellow, s)))
-    .on('add', copy)
-    .on('addDir', copy)
-    .on('change', copy)
-    .on('unlink', remove)
-    .on('unlinkDir', remove)
-    .on('error', e => console.log('[ERROR]'.red, e));
+if (watch) {
+	chokidar.watch(sourceGlobs, {
+		ignoreInitial: true
+	})
+		.on('ready', () => sourceGlobs.forEach(s => console.log('[WATCHING]'.cyan, s)))
```

Comment:
Why color chenged?

Reply:
I added some color diversity to the log messages to allow for easier reading. `COPY` is still yellow, I changed `WATCHING` to cyan, `DELETE` to red and added `CLEAN` in magenta.
I chose these colors (almost) randomly, if you want them changed just say the word.